### ISSUE TITLE
About Cosine similarity

### ISSUE THREAD

**Issue by fire5226** [hspark1212/chemeleon#6]
Thank you for the excellent work! I have two little ques:
1. Do you **only consider the cosine similarity** of the two embeddings (text and graph) when measuring the model performance in the contrastive learning?
2. How to **obtain the Heatmaps of cosine similarity in Fig. 2** because the map looks perfect. Do you use 'torch.nn.functional.cosine_similarity'? Is there any preprocessing for the embeddings (like normalization)? 

Thank you again for your work. Sorry to disturb you. Looking forward to a reply soon.

**Comment by hspark1212** [comment]
Hi @fire5226, thank you for your kind message.

1. Yes, I only use cosine similarity between the text and graph embeddings to evaluate the model performance in the contrastive learning setup.
2. For the heatmap in Figure 2, I use a custom cosine similarity implementation, but it should be equivalent to torch.nn.functional.cosine_similarity.
```
similarity = text_embeds @ graph_embeds.T
similarity = similarity / np.max(similarity)
similarity = similarity.clip(-1, 1)
```
I’m also sharing the Jupyter notebook so you can reproduce the figure directly. I hope it helps !
[vis_crystal_clip.ipynb.zip](https://github.com/user-attachments/files/20712939/vis_crystal_clip.ipynb.zip)

**Comment by fire5226** [comment]
> Hi [@fire5226](https://github.com/fire5226), thank you for your kind message.
> 
> 1. Yes, I only use cosine similarity between the text and graph embeddings to evaluate the model performance in the contrastive learning setup.
> 2. For the heatmap in Figure 2, I use a custom cosine similarity implementation, but it should be equivalent to torch.nn.functional.cosine_similarity.
> 
> ```
> similarity = text_embeds @ graph_embeds.T
> similarity = similarity / np.max(similarity)
> similarity = similarity.clip(-1, 1)
> ```
> 
> I’m also sharing the Jupyter notebook so you can reproduce the figure directly. I hope it helps ! [vis_crystal_clip.ipynb.zip](https://github.com/user-attachments/files/20712939/vis_crystal_clip.ipynb.zip)

Thank you very much!
